src/metrics/misc.ts

```diff
@@ -62,8 +66,9 @@ export class MetadataMetric implements Metric {
     [player: string]: number;
   };
 
-  constructor() {
+  constructor(chess) {
     this.clear();
+    this.chess = chess;
```

Comment:
fyi you could also convert the member methods you are using for board state into static methods and then you wouldn't need to modify the constructor like this

Reply:
any advantages to that approach? for now will keep as is